File: finder4j-backtrack-examples/src/main/java/com/github/K0zka/finder4j/backtrack/examples/queens/Main.java
package com.github.K0zka.finder4j.backtrack.examples.queens;

import com.github.K0zka.finder4j.backtrack.Backtrack;
import com.github.K0zka.finder4j.backtrack.SolutionListener;
import com.github.K0zka.finder4j.backtrack.TerminationStrategy;
import com.github.K0zka.finder4j.backtrack.listener.LoggerSolutionListener;
import com.github.K0zka.finder4j.backtrack.termination.AllSolutionsTerminationStrategy;

public class Main {
	public static void main(String[] args) {
		TerminationStrategy<QueensState> terminationStrategy = new AllSolutionsTerminationStrategy<QueensState>();
		final SolutionListener<QueensState, QueensStep> listener = new LoggerSolutionListener<QueensState, QueensStep>();
		Backtrack
				.backtrack(
						new QueensState(),
						new QueensStepFactory(),
						terminationStrategy,
						listener);
	}
}

File: finder4j-backtrack-examples/src/main/kotlin/com/github/k0zka/finder4j/backtrack/examples/queens/Main.kt
package com.github.k0zka.finder4j.backtrack.examples.queens

import com.github.k0zka.finder4j.backtrack.Backtrack
import com.github.k0zka.finder4j.backtrack.listener.LoggerSolutionListener
import com.github.k0zka.finder4j.backtrack.termination.AllSolutionsTerminationStrategy

object Main {
	@JvmStatic
	fun main(args: Array<String>) {
		val terminationStrategy = AllSolutionsTerminationStrategy<QueensState>()
		val listener = LoggerSolutionListener<QueensState>()
		Backtrack
				.backtrack(
						QueensState(),
						QueensStepFactory(),
						terminationStrategy,
						listener)
	}
}
